numpy/random/tests/test_random.py

```diff
@@ -352,40 +356,41 @@ def test_bytes(self):
         np.testing.assert_equal(actual, desired)
 
     def test_shuffle(self):
```

Comment:
From a comment by @pv https://github.com/scipy/scipy/pull/3719#discussion_r13781425 one reason is that it's harder to debug failing tests, for example

```python
import numpy as np
from numpy.testing import assert_allclose

def test_foo():
    A = np.arange(100)
    B = np.concatenate([A, A])
    yield assert_allclose, A, B
```

gives a failing test result that looks like this:

```
$ nosetests                           
F
======================================================================
FAIL: test.test_foo(array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
----------------------------------------------------------------------
Traceback (most recent call last):
[snip]
```

which looks messy.